streaming/src/main/scala/org/apache/spark/streaming/status/api/v1/StreamingApiRootResource.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.status.api.v1
+
+import javax.servlet.ServletContext
+import javax.ws.rs.{Path, WebApplicationException}
+import javax.ws.rs.core.{Context, Response}
+
+import org.eclipse.jetty.server.handler.ContextHandler
+import org.eclipse.jetty.servlet.{ServletContextHandler, ServletHolder}
+import org.glassfish.jersey.server.ServerProperties
+import org.glassfish.jersey.servlet.ServletContainer
+
+import org.apache.spark.status.api.v1.UIRoot
+import org.apache.spark.streaming.ui.StreamingJobProgressListener
+
+@Path("/v1")
```

Comment:
Hmmm. I don't like this very much. I'd change it as part of my ongoing work anyway, once this gets in, but might as well do that now.

The streaming API endpoints should show up under the existing API, instead of creating a new hierarchy. So you should have "/api/v1/application/:id/:attempt/streaming".

There's no streaming in the SHS now, but if there was, your current endpoint would look extremely out of place, and you wouldn't even have a way to reach the API for different applications (since this is not "mounted" under the applications resource).